components/analytics/bot-error-timeline.tsx

```
@@ -0,0 +1,309 @@
+
+    // Get priorities present in the data
+    const presentPriorities = useMemo(() => {
+        const priorities = new Set<string>()
+
+        errorsByDate.forEach(day => {
+            Object.keys(day.errorsByPriority).forEach(priority => {
+                if (day.errorsByPriority[priority] > 0) {
+                    priorities.add(priority)
+                }
+            })
+        })
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)